xarray/core/rolling.py

```diff
@@ -192,4 +191,76 @@ class DataArrayRolling(Rolling, ImplementsRollingArrayReduce):
     """Rolling object for DataArrays"""
 
 
+class DatasetRolling(Rolling, ImplementsRollingDatasetReduce):
+    """An object that implements the moving window pattern for Dataset.
+
+    This class inherites Rolling class, which is originally designed for
+    DataArray. Most of its methods also works for Dataset, but Dataset.concat()
+    used in reduce() gives undesired duplication of some DataArrays which does
+    not depend on the rolling-dim.
+
+    In this class, such DataArrays are kept aside as `self.fixed_ds`,
+    prior to call parent's __init__ method.
+    In `reduce` method, the fixed DataArrays are merged to the reduced
+    Dataset.
+
+    See Also
+    --------
+    Dataset.groupby
+    DataArray.groupby
+    Dataset.rolling
+    DataArray.rolling
+    """
+    def __init__(self, obj, min_periods=None, center=False, **windows):
+        """
+        Moving window object for Dataset.
+
+        Parameters
+        ----------
+        obj : Dataset
+            Object to window.
+        min_periods : int, default None
+            Minimum number of observations in window required to have a value
+            (otherwise result is NA). The default, None, is equivalent to
+            setting min_periods equal to the size of the window.
+        center : boolean, default False
+            Set the labels at the center of the window.
+        **windows : dim=window
+            dim : str
+                Name of the dimension to create the rolling iterator
+                along (e.g., `time`).
+            window : int
+                Size of the moving window.
+
+        Returns
+        -------
+        rolling : type of input argument
+        """
+        from .dataset import Dataset
+        dim, window = next(iter(windows.items()))
+        # DataArrays not depending on the rolling-axis are kept aside from obj
+        self.fixed_ds = Dataset({key: da for key, da in obj.data_vars.items()
```

Comment:
I will change the `DatasetRolling` structure to keep `data_var`s order.
My code destroys this order in `reduce` method by calling `ds.merge`.